layers/+lang/shell-scripts/packages.el

```diff
@@ -41,3 +43,18 @@
                    (string-match-p "\\.zsh\\'" buffer-file-name))
           (sh-set-shell "zsh")))
       (add-hook 'sh-mode-hook 'spacemacs//setup-shell))))
+
+
+(when (configuration-layer/layer-usedp 'auto-completion)
+  (defun python/post-init-company ()
+    (spacemacs|add-company-hook sh-mode)
+    (spacemacs|add-company-hook fish-mode))
+
+  (defun shell-scripts/init-company-shell ()
+    (use-package company-shell
+      :if (configuration-layer/package-usedp 'company)
+      :defer t
+      :init
+      (push 'company-shell      company-backends-sh-mode)
+      (push 'company-shell      company-backends-fish-mode)
+      (push 'company-fish-shell company-backends-fish-mode))))
```

Comment:
In company, only one backend is used at a time. You probably want
```el
(push '(company-shell company-fish-shell) company-backends-fish-mode)
```
i.e. a grouped backend.

Also wrap the the setup code in a `progn`.